### PR TITLE
if condition

### PR DESCRIPTION
**Pull request by Icermli**
as title

### CODE REVIEW

**Review by ncying — CHANGES_REQUESTED** [review]
LGTM. Some additional requests, all other OPCs unit test tested the particular functions, it is fine. However, here if we only test the `executeOpcBlock` and `runCondition `, we may miss the logic of `If` and `IfElse`, since here the two OPCs are combined with two functions listed above. So I prefer to add extra tests that directly use `IfOpcDiff. parseBytes`.

**Comment by Icermli** [comment]
Sure, I will try to add that test

File: src/test/scala/vsys/blockchain/state/opcdiffs/IfOpcDiffTest.scala [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          DataEntry(Shorts.toByteArray(0), DataType.OpcBlock)))))
+    IfOpcDiff.parseBytes(executionContext)(
+      Array[Byte](2.toByte, 0.toByte, 1.toByte, 2.toByte),
+      Seq[DataEntry](DataEntry(Array(0.toByte), DataType.Boolean),
+        DataEntry(Shorts.toByteArray(0), DataType.OpcBlock),
+        DataEntry(Shorts.toByteArray(0), DataType.OpcBlock))) should be (
+      Right((OpcDiff(Map(),Map(),Map(),Map(),Map(),Map(),Map(),Map()),
+        List(DataEntry(Array(0.toByte), DataType.Boolean),
+          DataEntry(Shorts.toByteArray(0), DataType.OpcBlock),
+          DataEntry(Shorts.toByteArray(0), DataType.OpcBlock)))))
```
Comment (ncying) [inline src/test/scala/vsys/blockchain/state/opcdiffs/IfOpcDiffTest.scala:82]: What's the difference here between if and else part?

Comment (Icermli) [reply src/test/scala/vsys/blockchain/state/opcdiffs/IfOpcDiffTest.scala:82]: no difference in `OpcBlock`

Comment (ncying) [reply src/test/scala/vsys/blockchain/state/opcdiffs/IfOpcDiffTest.scala:82]: you can design and let else part return a validation error~ = =\\

Comment (Icermli) [reply src/test/scala/vsys/blockchain/state/opcdiffs/IfOpcDiffTest.scala:82]: ok

**Review by ncying — APPROVED** [review]

**Pull request merged**